docs/t-sql/functions/replace-transact-sql.md

```diff
@@ -92,6 +92,28 @@ This is a desk
 (1 row(s) affected)  
 ```  
 
+The following example calculates the number of spaces in a sentence using the `REPLACE` function. First, it calculates the length of the sentence with the `LEN` function. It then replaces the ' ' characters with '' with REPLACE . After this process, it calculates the length of the sentence again. The resulting difference is the number of space characters in the sentence.
+
+```sql  
+DECLARE @str NVARCHAR(100), @len1 INT, @len2 INT
+SET @str = 'This is a sentence with a space in it.'
+SET @len1 = LEN(@str)
+SET @str = REPLACE(@str, ' ', '')
+SET @len2 = LEN(@str)
+SELECT 'Number of spaces in the sentence:' + CONVERT(NVARCHAR, @len1 - @len2)

```

Comment:
Official documentation examples should be consistent and follow best practices:

1. When working with `NVARCHAR` data, string literals should be prefixed with `N`.
2. Statements should be terminated with semicolons (this isn't currently enforced, but not doing so is at least officially deprecated).
3. When using variable-length data types, please always specify a length, even if the default 30 works, because in some scenarios the default is actually 1. Also, not everyone knows that the default is 30.

```suggestion
DECLARE @str NVARCHAR(100), @len1 INT, @len2 INT;
SET @str = N'This is a sentence with a space in it.';
SET @len1 = LEN(@str);
SET @str = REPLACE(@str, N' ', N'');
SET @len2 = LEN(@str);
SELECT N'Number of spaces in the string: ' + CONVERT(NVARCHAR(20), @len1 - @len2);

```

Also, in the final line I added a space after the colon for readability/formatting and changed "sentence" to be "string" as this could just as easily be used for multiple sentences.